src/utils/dataPlane-utils.ts

```diff
@@ -139,6 +140,85 @@ export const getJournals = async (
         brokerToken
     );
 
+export const DATA_PLANE_PREFIX = 'ops/dp/';
+export const PUBLIC_DATA_PLANE_PREFIX = `${DATA_PLANE_PREFIX}public/`;
+
+export const getDataPlaneScope = (
+    dataPlaneName: string
+): DataPlaneOption['scope'] => {
+    return dataPlaneName.startsWith(PUBLIC_DATA_PLANE_PREFIX)
+        ? 'public'
+        : 'private';
+};
+
+const splitTruncatedDataPlaneName = (
+    dataPlaneName: string,
+    basePrefix: string
+) => {
+    const truncatedName = dataPlaneName.substring(basePrefix.length);
+
+    const slashIndex = truncatedName.lastIndexOf('/');
+
+    const prefix =
+        slashIndex === -1 ? '' : truncatedName.substring(0, slashIndex + 1);
+
+    const suffix =
+        slashIndex === -1
+            ? truncatedName
+            : truncatedName.substring(slashIndex + 1);
+
+    return [prefix, suffix];
+};
+
+const splitDataPlaneSuffix = (suffix: string, firstHyphenIndex: number) => {
```

Comment:
This is good for now as it is super understandable... but this seems like the rare case possibly splitting up the string with a RegEx could be easier.

Reply:
Left a comment behind for this.